R/bambu-extendAnnotations-utilityExtend.R

```diff
@@ -674,20 +676,19 @@ isore.estimateDistanceToAnnotations <- function(seReadClass,
     distTable <- left_join(distTable, select(readClassTable,
                                              readClassId, confidenceType), by = "readClassId") %>%
     mutate(relativeReadCount = readCount / txNumberFiltered)
-  distTable <- dplyr::select(distTable, annotationTxId, readClassId,
-                             readCount, compatible, equal,dist)
-  distTable <- left_join(distTable, as_tibble(mcols(annotationGrangesList)[,
-                                                                           c("TXNAME", "GENEID")]),by = c("annotationTxId" = "TXNAME"))
+  distTable <- dplyr::select(distTable, annotationTxId, txid, readClassId,
+      readCount, compatible, equal,dist)
+  distTable <- left_join(distTable, as_tibble(mcols(annotationGrangesList)[, c("txid", "GENEID")]),
+      by = c("txid" = "txid"))
+  
   end.ptm <- proc.time()
   if (verbose) message("calculated distance table in ",
                        round((end.ptm - start.ptm)[3] / 60, 1), " mins.")
   readClassTable <- addGeneIdsToReadClassTable(readClassTable, distTable, 
                                                seReadClass, verbose)
-  
   distTable <- DataFrame(distTable)
-  matchIds <- match(distTable$annotationTxId,mcols(annotationGrangesList)$TXNAME)
-  distTable$txid <- mcols(annotationGrangesList)$txid[matchIds]
-  distTable$eqClassById <- mcols(annotationGrangesList)$eqClassById[matchIds]
+  distTable$eqClassById <- as.list(mcols(annotationGrangesList)$eqClassById)[distTable$txid]
```

Comment:
is the as.list() required here?

Reply:
yes, this is to change the type of the eqClassbyId in mcols(annotations) from S4 to S3, so it can be used later in dplyr for grouping or other things.